lib/ravelin/event.rb

```diff
@@ -75,13 +75,11 @@ def payload_customer_reference_present?
       self.payload.keys.any? {|k| %i(customer_id temp_customer_id).include? k }
     end
 
-    def convert_to_epoch(time)
-      if time.is_a?(Time)
-        time.to_i
-      elsif time.is_a?(DateTime) && time.respond_to?(:to_i)
-        time.to_i
-      elsif time.is_a?(Integer)
-        time
+    def convert_to_epoch(val)
+      if val.is_a?(Time) || val.is_a?(Date) || val.is_a?(DateTime)
```

Comment:
I wanted to make sure it was just times, dates and integers, otherwise users might enter strings or nil.

* `nil.to_i` gives `0` which in January 1st, 1970
* `"2016-02-24 12:31:32 +0000".to_i` gives 2016